osu.Game.Rulesets.Taiko/Mods/TaikoModSimplified.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Bindables;
+using osu.Framework.Localisation;
+using osu.Game.Beatmaps;
+using osu.Game.Beatmaps.ControlPoints;
+using osu.Game.Configuration;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Taiko.Beatmaps;
+using osu.Game.Rulesets.Taiko.Objects;
+
+namespace osu.Game.Rulesets.Taiko.Mods
+{
+    public class TaikoModSimplified : Mod, IApplicableToBeatmap
+    {
+        public override string Name => "Simplified";
+        public override string Acronym => "SF";
+        public override double ScoreMultiplier => 0.6;
+        public override LocalisableString Description => "Simplify tricky rhythms!";
+        public override ModType Type => ModType.DifficultyReduction;
+
+        [SettingSource("1/3 to 1/2 conversion", "Converts 1/3 patterns to 1/2 rhythm.")]
+        public Bindable<bool> OneThirdConversion { get; } = new BindableBool(false);
+
+        [SettingSource("1/6 to 1/4 conversion", "Converts 1/6 patterns to 1/4 rhythm.")]
+        public Bindable<bool> OneSixthConversion { get; } = new BindableBool(true);
+
+        [SettingSource("1/8 to 1/4 conversion", "Converts 1/8 patterns to 1/4 rhythm.")]
+        public Bindable<bool> OneEighthConversion { get; } = new BindableBool(false);
+
+        public void ApplyToBeatmap(IBeatmap beatmap)
+        {
+            var taikoBeatmap = (TaikoBeatmap)beatmap;
+            var controlPointInfo = taikoBeatmap.ControlPointInfo;
+            List<Hit> toRemove = new List<Hit>();
+
+            // Snap conversions for rhythms
+            var snapConversions = new Dictionary<int, int>
+            {
+                { 8, 4 }, // 1/8 snap to 1/4 snap
+                { 6, 4 }, // 1/6 snap to 1/4 snap
+                { 3, 2 }, // 1/3 snap to 1/2 snap
+            };
+
+            bool inPattern = false;
+
+            List<Hit> hits = taikoBeatmap.HitObjects.Where(obj => obj is Hit).Cast<Hit>().ToList();
```

Comment:
I have a feeling that doing this is incorrect, because doing something like placing a drum roll on a 1/1 beat after a 1/3rds rhythm is going to result the last note of the 1/3rds remaining snapped to 1/3. In other words, discounting non-hits this early essentially deletes non-hits from the map, skewing time intervals between objects and potentially ending patterns prematurely.

Reply:
I think you might be misunderstanding this.

the way the mod works is by reading patterns in groups of 3, then adjusting every 2nd note to form a 1/4 rhythm and removing the 3rd note entirely, with the mod being blind to drumrolls/swells.

the latter was a deliberate decision because having them account for conversion would result in unintended consequences, namely them getting deleted or missnapped.

here's the only 3 outcomes that can happen involving them right now:

### Case 1

| before | after |
| :-: | :-: |
| ![image](https://github.com/user-attachments/assets/d77a9862-33fd-4119-8b50-b7803ea8a0ee) |![image](https://github.com/user-attachments/assets/d77a9862-33fd-4119-8b50-b7803ea8a0ee) |

the pattern will be untouched which is fine because snaps are not much relevant with drumrolls/swells compared to notes playability-wise.

---

### Case 2

| before | after |
| :-: | :-: |
| ![image](https://github.com/user-attachments/assets/a432edf2-7c44-46d7-9cf4-e368309aed80) | ![image](https://github.com/user-attachments/assets/09b5b954-8a93-415d-b6c5-d92ae54eb475) |

the 2nd don will be moved closer to the drumroll/swell, which is arguably fine for the aforementioned reason.

---

### Case 3

| before | after |
| :-: | :-: |
| ![image](https://github.com/user-attachments/assets/0c4bb76a-90f1-43cf-9cb1-c23562fbeaaf) | ![image](https://github.com/user-attachments/assets/4a7c5713-62ea-44a0-b051-da84c27ecbcd) |

the 2nd note will be adjusted accordingly, and the 3rd note would be removed, as expected. playability around the drumroll/swell is unaffected